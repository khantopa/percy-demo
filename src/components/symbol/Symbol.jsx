import React from 'react';
import PropTypes from 'prop-types';

import './Symbol.scss';


const Symbol =(props) => {
  const { symbol, color } = props;
  const imgUrl = `../../../assets/${symbol}.png`;
  const style = {
    backgroundColor: color,
    backgroundImage: `url(${imgUrl})`,
  }
  return <div className="symbol" style = { style } ></div>
}


Symbol.defaultProps = {
  symbol: "",
  color: "#D8EEF2"
};


Symbol.propTypes = {
  color: PropTypes.string.isRequired,
  symbol: PropTypes.string.isRequired
}


export default Symbol;


