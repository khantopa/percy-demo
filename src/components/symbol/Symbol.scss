@import '../../styles/main.scss';


.symbol { 
  box-sizing: border-box;
  background: $primary-light center/60% no-repeat;
  height: 100%;
  width: 100%;
  border: 3px solid $light;
  border-radius: 50%;
  box-shadow: inset 0 5px 10px 4px rgba(0, 0, 0, .35);
}