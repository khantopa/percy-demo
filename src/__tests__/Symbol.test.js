import Symbol from '../components/Symbol';

let wrapper, props;


beforeEach(() => {
  props = {} ; 
  wrapper =shallow(<Symbol player = { props } />);
});

describe('<Symbol>', () => {
  
  it('should <Board> defined', () => {
    expect(wrapper).toBeDefined();
  });

  it('renders <Symbol> without crashing', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

});
