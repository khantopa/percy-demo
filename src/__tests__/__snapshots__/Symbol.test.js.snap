// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Symbol> renders <Symbol> without crashing 1`] = `
<div
  className="symbol"
  style={
    Object {
      "backgroundColor": "#D8EEF2",
      "backgroundImage": "url(../../../assets/.png)",
    }
  }
/>
`;
